### ISSUE TITLE
Spotify support confusing

### ISSUE THREAD

**Issue by natedogith1** [ytdl-org/youtube-dl#31037]
## Checklist

- [x] I'm reporting a broken site support
- [ ] I've verified that I'm running youtube-dl version **2021.12.17**
- [ ] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Description

[supportedsites.md](https://github.com/ytdl-org/youtube-dl/blob/master/docs/supportedsites.md) says spotify is supported, but it fails with the typical open.spotify.com links (related issues: #30180, #29722, #28660).  The documentation should be updated to make it clearer that this use case isn't supported.

**Comment by 89z** [comment]
I think its possible to get these. Do you have example URL?

**Comment by dirkf** [comment]
Spotify is supported. The extractor knows about `/episode/` and `/show/` URLs.

Otherwise, duplicate of #29722, as you have identified.

The yt-dlp extractor also finds URLS with additional initial `embed` or `embed-podcast` before the URL type (as above) and this could easily be back-ported.

Otherwise, is it worth adding a special `SpotifyUnsupportedIE` to report on unsupported Spotify URL patterns?

**Comment by rautamiekka** [comment]
> Otherwise, is it worth adding a special `SpotifyUnsupportedIE` to report on unsupported Spotify URL patterns?

**Let's put it this way:** is the use case of "_prevent the endless stream of ticket dupes and uncertainty that can only be confirmed by spending a very boring hour or 2 reading both the official Spotify docs and the ytdl source code ?_" worth it ?

**Comment by gamer191** [comment]
> Otherwise, is it worth adding a special `SpotifyUnsupportedIE` to report on unsupported Spotify URL patterns?

I once started working on an yt-dlp IE [here](https://github.com/gamer191/yt-dlp/tree/knowndrmie) to report a warning whenever the user inputs a DRM-protected url, and then fall back on the generic extractor. Currently it's abandoned, but it could probably be used as a template for a bigger `UnsupportedSitesIE`, which could include the Spotify unsupported IE.

Now that I think about it, I might continue working on it some time soon.

**Comment by gamer191** [comment]
Wait, do you still want to work on it @sulyi?

**Comment by sulyi** [comment]
> want to work on the DrmProtectedIE

That was someone else's idea, I just joked about implementing it. It's more of a in-fracture thing which is my jam.
#discord

**Comment by KarlRW** [comment]
Just to clear this up, are things like the JRE podcast possible to download? It's available for free. Trying to download the episodes because[ spotify has so many issues](https://www.google.com/search?q=spotify+glitchy+broken+reddit) it's become unusable for me.

**Comment by dirkf** [comment]
Your open.spotify.com link is marked as "extractor not working" in yt-dlp and gives "Unsupported URL" in yt-dl.

**Comment by gamer191** [comment]
> https://open.spotify.com/embed/episode/4OXGSwuHYf0gHtJxGWIbLL/video?utm_source=generator

This link works after disabling DRM, so it would be possible to add support for

Whether it would be merged is another question. I'd be interested @pukkandan's take on that question, since he's generally opposed to Spotify support, due to Spotify's history of legal action, but that particular video also plays DRM-free without any workaround. I'd also love to hear @dirkf's opinion (and while we're at it, @dirkf's opinion on https://github.com/yt-dlp/yt-dlp/pull/4935, which wasn't merged due to [Deezer's DMCA history](https://github.com/github/dmca/blob/master/2021/02/2021-02-10-deezer.md), along with some licensing issues)

**Comment by pukkandan** [comment]
> I'd be interested @pukkandan's take on that question

https://github.com/yt-dlp/yt-dlp/pull/4935#issuecomment-1460160547

**Comment by 4cq2** [comment]
since this is the YouTube-DL issue tracker, and not YT-DLP, consider my comments in that context.

I agree generally with a previous comment that any Spotify non DRM options should be downloadable with YouTube-DL. If you can do it anonymous with a web browser without DRM, then Spotify (or anyone else) doesn't have a let to stand on in regards to any DMCA claim. Yes it might be annoying worrying about legal action, but thats a slippery slope. before you know it, you're running away from anything that even smells funny.

regarding Deezer, same story. if we are going to refuse Deezer support, then we might as well pull all HLSe support too, as its basically the same thing. as far as I am concerned, anything except proper DRM (Widevine, FairPlay etc) is fair game. Bring it on. I will gladly contribute to YouTube-DL legal fund should a case be presented.

**Comment by dirkf** [comment]
My feeling is that effort is unlikely to be devoted to sites where little of the content is accessible, since it will only lead to disappointed users, and that such effort could be used on more useful extractors.

Better to person up and admit that some other technique, maybe involving a microphone or a headphone socket in this case, would be more generally effective.

**Comment by 4cq2** [comment]
> Better to person up and admit that some other technique, maybe involving a microphone or a headphone socket in this case, would be more generally effective.

not really. Widevine L3 (which Spotify uses) has been reversed for over a year now (by me and others), which allows a byte for byte reproduction, which goes without saying is much better than any analog method.

**Comment by 4cq2** [comment]
https://github.com/ytdl-org/youtube-dl/issues/32362

**Comment by 4cq2** [comment]
note the provided example is only a preview, it seems the full media is hidden behind a login. if someone can provide me with an account for testing I should be able to code this.

**Comment by gamer191** [comment]
> Better to person up and admit that some other technique, maybe involving a microphone or a headphone socket in this case, would be more generally effective.

or possibly https://github.com/yt-dlp/yt-dlp/issues/1368#issuecomment-955783122, but that's legally quite risky, especially since I'm pretty sure ZSpotify got DMCAed

**Comment by dirkf** [comment]
It wasn't very effective if so, as many heads of the ZSpotify Hydra remain. But also GPL.

**Comment by 4cq2** [comment]
unsubscribing from this thread, as it seems to have fallen victim to the typical crap. no decent examples, no accounts, babies crying about DRM. at me or contact me privately if something changes. peace.

**Comment by gamer191** [comment]
> It wasn't very effective if so, as many heads of the ZSpotify Hydra remain.

Um, what?

> unsubscribing from this thread, as it seems to have fallen victim to the typical crap. no decent examples, no accounts, babies crying about DRM. at me or contact me privately if something changes. peace.

I was unaware that pointing out that sending instructions to workaround Spotify's DRM counted as "babies crying about DRM". I guess I learn something new everyday!

**Comment by 4cq2** [comment]
ah yes, the great @gamer191. where would the programming world be without your great repos:

https://github.com/gamer191?tab=repositories&type=source

**Comment by gamer191** [comment]
@dirkf while we're discussing DRM, what's your opinion on [yt-dlp/yt-dlp#4935](https://github.com/yt-dlp/yt-dlp/pull/4935), which wasn't merged due to [Deezer's DMCA history](https://github.com/github/dmca/blob/master/2021/02/2021-02-10-deezer.md) (that PR was rejected anyway, because it's allegedly been plagurised and hence doesn't fall under unlicense, but that's not the main issue, and it can be solved by rewriting the pr)

**Comment by 1268** [comment]
> which wasn't merged due to [Deezer's DMCA history](https://github.com/github/dmca/blob/master/2021/02/2021-02-10-deezer.md)

I wish people would stop with the scare tactics. that request was filed two years ago. since then plenty of implementations still exist in the wide open:

https://github.com/search?q=deezer+pushed:>2022-07-16+stars:>9

theres caution, and then theres just silliness.

**Comment by eflister** [comment]
spotify is listed as supported, but elsewhere in the issue tracker it is claimed not to be because of drm.  which is it?  can the doc be updated to clearly state whether it is supported?  if so, support playlists please.  https://github.com/ytdl-org/youtube-dl/issues/29722#issuecomment-1853262541

**Comment by dirkf** [comment]
>[The extractor only supports podcasts. Music on spotify is DRM protected and cannot be supported.](https://github.com/ytdl-org/youtube-dl/issues/29722)
